protocols/v2/codec-sv2/src/lib.rs

```diff
@@ -34,20 +68,74 @@ pub use buffer_sv2;
 
 pub use framing_sv2::{self, framing::handshake_message_to_frame as h2f};
 
+/// Represents the role in the Noise handshake process, either as an initiator or a responder.
+///
+/// The Noise protocol requires two roles during the handshake process:
+/// - **Initiator**: The party that starts the handshake by sending the initial message.
+/// - **Responder**: The party that waits for the initiator's message and responds to it.
+///
+/// This enum distinguishes between these two roles, allowing the codec to handle the handshake
+/// process accordingly.
+#[allow(clippy::large_enum_variant)]
+#[cfg(feature = "noise_sv2")]
+#[derive(Debug)]
+pub enum HandshakeRole {
+    /// The initiator role in the Noise handshake process.
+    ///
+    /// The initiator starts the handshake by sending the initial message. This variant stores an
+    /// `Initiator` object, which contains the necessary state and cryptographic materials for the
+    /// initiator's part in the Noise handshake.
+    Initiator(Box<noise_sv2::Initiator>),
+
+    /// The responder role in the Noise handshake process.
+    ///
+    /// The responder waits for the initiator's handshake message and then responds. This variant
+    /// stores a `Responder` object, which contains the necessary state and cryptographic materials
+    /// for the responder's part in the Noise handshake.
+    Responder(Box<noise_sv2::Responder>),
+}
+
+/// Represents the state of the Noise protocol codec during different phases: initialization,
+/// handshake, or transport mode, where encryption and decryption are fully operational.
+///
+/// The state transitions from initialization (`NotInitialized`) to handshake (`HandShake`)
+/// and finally to transport mode (`Transport`) as the encryption handshake is completed.
```

Comment:
Variables here and else where are not linked correctly and not consistently. For example (`Transport`) is referenced is different ways that also usually dont link to the actual object. 

If we leave the variables like this, in docs.rs the variables wont be clickable(they will be gray instead of blue).


I suggest to change it to the following:
```suggestion
/// and finally to transport mode [`State::Transport`] as the encryption handshake is completed.
///
/// [`State::Transport`]: crate::State::Transport
```